server/src/Ogmios/Parser.hs

```diff
@@ -0,0 +1,146 @@
+module Ogmios.Parser (decodeProtocolParameters) where
+
+import Cardano.Api (
+  AnyPlutusScriptVersion (AnyPlutusScriptVersion),
+  CostModel,
+  ExecutionUnitPrices (ExecutionUnitPrices),
+  ExecutionUnits,
+  Lovelace (Lovelace),
+  PlutusScriptVersion (PlutusScriptV1),
+ )
+import Cardano.Api.Shelley (ProtocolParameters (ProtocolParameters))
+
+import Data.Text qualified as Text
+import Text.Parsec qualified as Parsec
+import Text.Parsec.Char qualified as Parsec.Char
+import Text.ParserCombinators.Parsec.Combinator (many1)
+
+import Data.Aeson qualified as Aeson
+import Data.Aeson.BetterErrors (
+  Parse,
+  asIntegral,
+  displayError,
+  fromAesonParser,
+  key,
+  parseValue,
+  perhaps,
+  withString,
+  (<|>),
+ )
+import Data.ByteString.Lazy (ByteString)
+import Data.Char (isDigit)
+import Data.Map qualified as Map
+import Data.Ratio ((%))
+import GHC.Natural (Natural, naturalFromInteger)
+
+parseVersion :: Parse e (Natural, Natural)
+parseVersion =
+  key "protocolVersion" $
+    (,) <$> parseNatural "major" <*> parseNatural "minor"
+
+parseNatural :: Text.Text -> Parse e Natural
+parseNatural strKey =
+  naturalFromInteger <$> key strKey (asIntegral :: Parse e Integer)
+
+parseLovelace :: Text.Text -> Parse e Lovelace
+parseLovelace strKey =
+  key strKey $
+    Lovelace <$> (asIntegral :: Parse e Integer)
+
+parseRational :: Text.Text -> Parse Text.Text Rational
+parseRational strKey =
+  key strKey (withString rationalParser)
+
+parseExecutionPrices :: Parse Text.Text (Maybe ExecutionUnitPrices)
+parseExecutionPrices =
+  perhaps $
+    ExecutionUnitPrices
+      <$> parseRational "steps"
+      <*> parseRational "memory"
+
+parseExecutionUnits :: Parse e (Maybe ExecutionUnits)
+parseExecutionUnits = fromAesonParser
+
+parseCostModels :: Parse e (Map.Map AnyPlutusScriptVersion CostModel)
+parseCostModels =
+  Map.singleton (AnyPlutusScriptVersion PlutusScriptV1)
+    <$> key "plutus:v1" fromAesonParser
+
+parseResult :: Parse Text.Text ProtocolParameters
+parseResult =
+  key "result" $
+    ProtocolParameters
+      <$> parseVersion
+      <*> parseRational "decentralizationParameter"
+      -- TODO : How to parse the value `neutral` from ogmios?
```

Comment:
That should be addressed in this PR I think

Reply:
yep, I stated it at the top but now is solved.

fixed.